runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -517,8 +518,20 @@ public synchronized void invalidateLayout() {
                         // it is acceptable (at least the code on 10/13/2016 does not have issues)
                         // but setEpoch of routers needs to be synchronized as those variables are
                         // not local.
-                        l.getAllServers().stream().map(getRouterFunction).forEach(x ->
-                                x.setEpoch(l.getEpoch()));
+                        try {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)